docs/types.rst

```diff
@@ -751,13 +751,12 @@ Members
 ^^^^^^^
 
 **length**:
-    Arrays have a ``length`` member to hold their number of elements.
-    Dynamic arrays can be resized in storage (not in memory) by changing the
-    ``.length`` member. This does not happen automatically when attempting to access elements outside the current length. The size of memory arrays is fixed (but dynamic, i.e. it can depend on runtime parameters) once they are created.
+    Fixed-length arrays (storage or memory) have a read-only ``length`` member to access their number of elements.
+    Dynamic-sided arrays (only available for storage) have a read-write ``length`` member to resize the arary. Increasing the length adds unitialized elements to the array, this is O(1) complexity. Reducing the length performs ``delete()`` on each removed element and is O(n) complexity proportional to the number of elements being deleted. Please note that calling ``length--`` on an empty array will set the length of the array to 2^256-1 due to uint256 underflow wrapping.
```

Comment:
The spellchecker caught `unitialized  ==> uninitialized` here.

Reply:
There are other typos, such as:
`dynamic-sided => dynamically-sized`
`arary => array`

Reply:
You can also have `memory` dynamically-sized arrays, just not resize them that way.

Reply:
Fixed typos here in a commit with a message that appropriately, but unintentionally, includes the same 2b1502f22bfb992ff61de12619f0d7c2526ff5f1

Reply:
Are you talking about memory arrays that have a length specified in runtime?

Reply:
Yes, we usually call them dynamically-sized arrays, even if you cannot change their length once created.